client/src/components/ForgotPasswordForm/ForgotPasswordForm.js

```diff
@@ -1,62 +1,108 @@
+      page: 1,
+      email: '',
+      code: '',
+      password: ''
+    }
+  }
 
-function ForgotPasswordForm (props) {
-  const { handleSubmit, sendValidationCode } = props
-  return (
-    <form onSubmit={ handleSubmit(sendValidationCode) }
-      className={styles.form}>
-
-      <h1
-        className={styles.title}>
-          Forgot Password?
-      </h1>
-
-      <h3>{"Let's get you a new one!"}</h3>
-
-      <Field
-        name="email"
-        component={EmailField}/>
-
-      <button
-        className={styles.buttonSubmit}
-        type="submit">
-          Send Verification Code
-      </button>
-    </form>
-  )
-}
+  onChange (event) {
+    this.setState({[event.target.name]: event.target.value})
+  }
+
+  nextPage ({email}) {
+    return this.props.sendValidationCode(email).then(() => {
+      this.setState({page: this.state.page + 1})
+    }).catch((error) => { throw new SubmissionError({ code: error.name, _error: error.message }) })
+  }
+
+  async validate (values, dispatch) {
+    const _error = 'Incorrect code, please try again!'
+    const apiOptions = apiOptionsFromState()
+    const response = await forgotPasswordApiClient.validateCode(apiOptions, this.props.email, values.code)
+    if (response.status === 200) {
+      this.setState({page: this.state.page + 1})
+    } else {
+      throw new SubmissionError({ code: response.statusText, _error })
+    }
+  }
 
-EmailField.propTypes = {
-  input: PropTypes.object
+  changePass (values, dispatch) {
+    return this.props.changePassword(this.props.email, values.password).then(() => {
+      dispatch(push('/'))
+    }).catch(error => {
+      throw new SubmissionError({code: error.name, _error: error.message})
+    })
+  }
+
+  render () {
+    const {page} = this.state
+    const {error} = this.props
+    return (
+      <div>
+        {page === 1 && <SendVerificationCodeForm email={this.state.email} onChangeEmail={this.onChange} onSubmit={this.nextPage} />}
+        {page === 2 &&
+        <div>
+          <ValidateVerificationCode code={this.state.code} onChangeCode={this.onChange} onSubmit={this.validate}/>
+        </div>}
+        {page === 3 &&
+        <div>
+          <NewPasswordForm password={this.state.password} onChangePassword={this.onChange} onSubmit={this.changePass}/>
+        </div>}
+        <div className={styles.errorContent}>
+          {error}
+        </div>
+      </div>
+
+    )
+  }
 }
+
 ForgotPasswordForm.propTypes = {
+  sendValidationCode: PropTypes.func,
+  validateCode: PropTypes.func,
+  email: PropTypes.string,
   handleSubmit: PropTypes.func,
-  sendValidationCode: PropTypes.func
+  error: PropTypes.any,
+  changePassword: PropTypes.func
 }
 
 const mapDispatchToProps = {
-  sendValidationCode: AuthActionCreator.sendValidationCode
+  sendValidationCode: AuthActionCreator.sendValidationCode,
+  changePassword: AuthActionCreator.changePassword
+}
+
+function mapStateToProps (state) {
+  return {
+    email: state.forgotPass.email
```

Comment:
With these latest changes, can we remove this `mapStateToProps` function and the forgotPasswordReducer?

Reply:
I can do that change later...removing the reducer makes the action unnecessary and the code that is done there can be on the component since I'm not dispatching anything.